library/std/src/sys/unix/fd/tests.rs

```diff
@@ -3,7 +3,16 @@ use core::mem::ManuallyDrop;
 
 #[test]
 fn limit_vector_count() {
-    let stdout = ManuallyDrop::new(unsafe { FileDesc { fd: 1 } });
+    let stdout = ManuallyDrop::new(
+        #[cfg(not(target_os = "fuchsia"))]
```

Comment:
I wasn't sure if this was needed or not. Is the unsafe caused by the rustc_layout attrs or is it done manually?

Reply:
Looks like it is caused by the rustc_layout attributes. nit: A way to avoid repetition would be

```rust
#[cfg_attr(target_os = "fuchsia", allow(unused_unsafe))]
```

above the unsafe expression.

Reply:
Done